### PR TITLE
Handle new _serialize === true functionality

### PR DESCRIPTION
**Pull request by Zuluru**
Resolution for issue #10

### CODE REVIEW

File: src/Controller/Component/AjaxComponent.php [hunk 1]
```diff
@@ -88,11 +88,14 @@ protected function _respondAsAjax() {
...
-			$serializeKeys = array_merge($serializeKeys, $this->Controller->viewVars['_serialize']);
+		// If _serialize is true, *all* viewVars will be serialized; no need to add _message.
+		if ($this->Controller->viewVars['_serialize'] !== true) {
+			$serializeKeys = ['_message'];
+			if (!empty($this->Controller->viewVars['_serialize'])) {
+				$serializeKeys = array_merge($serializeKeys, $this->Controller->viewVars['_serialize']);
+			}
+			$this->Controller->set('_serialize', $serializeKeys);
 		}
-		$this->Controller->set('_serialize', $serializeKeys);
```
Comment (dereuromark) [inline src/Controller/Component/AjaxComponent.php]: It would be cleaner and easier to just return early if (empty())

Comment (Zuluru) [inline src/Controller/Component/AjaxComponent.php]: I tend to agree, but so many people seem obsessed with having only a single return point from functions. I'll make the change and submit a new PR?

Comment (dereuromark) [inline src/Controller/Component/AjaxComponent.php]: Having return early is one of the most useful code practices. I dont know any sane person who would argue against that.

Comment (dereuromark) [inline src/Controller/Component/AjaxComponent.php]: No need, you can just update this PR with a 2nd commit.

**Comment by dereuromark** [comment]
See, so much cleaner :) Beautiful, thx!
A test case would be nice, but aint necessary I guess.

**Comment by dereuromark** [comment]
PS: Tests are failing, you need an additional `!empty($this->Controller->viewVars['_serialize'])` check.

**Comment by Zuluru** [comment]
In the interests of reducing duplicate code, I'm tempted to put the new block inside the existing `!empty` condition. Seems the code could be less clear that way though. Got a preference?

**Comment by dereuromark** [comment]
return early still trumps it IMO as less levels of nesting is always worse.
But either way works.

**Pull request merged**